elastic/kibana
# [Advanced settings] Add json validation to code editor field

Closes https://github.com/elastic/kibana/issues/175728

## Summary

This PR adds JSON syntax validation to the code editor field, so that the field components that don't use server-side `schema` validation (such as the settings in APM) are also validated against JSON syntax.

**How to test:**
1. Start Es with `yarn es snapshot` and Kibana with `yarn start`.
2. Go to Observability -> APM -> Settings
3. Type in some invalid JSON in the `observability:apmAWSLambdaPriceFactor` field and verify that an error message is diplayed.

<img width="1178" alt="Screenshot 2024-01-29 at 11 19 35" src="https://github.com/elastic/kibana/assets/59341489/eae01ee4-8503-4a36-ad09-1288abcd5142">
